### ISSUE TITLE
Add support for BLE HID keyboard

### ISSUE THREAD

**Issue by nanosonde** [esphome/feature-requests#1450]
**Describe the problem you have/What new integration you would like**

Implement a BLE HID Keyboard to send keystrokes to connected tablets or smartphones.

**Please describe your use case for this integration and alternatives you've tried:**

Use-case: Wall-mounted iPad which is used as a smart home (status) display which is supposed to be woken up by sending for example ESC key to wake it up from sleep. OR: control the global screenbrightness by sending keystrokes via BLE HID keyboard.
All this is not really possible from an app itself as iOS does not support this easily.

I have already done this with my quick and dirty Arduino sketch using an ESP32 and my iPad. It is working as expected, but I would like to see this in ESPHome too.

**Additional context**

Basically most of this code could be used: https://github.com/olegos76/nimble_kbdhid_example
It also works with the older ESP-IDF 3.3.4.

**Comment by nanosonde** [comment]
It seems that this has already been done: 
- for ESPHome: https://github.com/bburky/esphome-projects
- for Homey: https://github.com/satoer/HomeyESP32BluetoothKeyboard

See also this thread: https://community.home-assistant.io/t/feature-request-using-esphome-to-emulate-a-usb-keyboard/150706/14

**Comment by ThisIsTheOnlyUsernameAvailable** [comment]
I'm working on a DIY replacement for Harmony, so this would be ideal for native support within ESPHome.

**Comment by tomaszduda23** [comment]
I'm working on keyboard support. For now it is only USB https://github.com/esphome/feature-requests/issues/1918

**Comment by tomaszduda23** [comment]
BLE keyboard https://github.com/tomaszduda23/esphome/pull/1
